api/types/src/move_types.rs

```diff
@@ -382,6 +383,20 @@ pub struct MoveStructTag {
     pub generic_type_params: Vec<MoveType>,
 }
 
+impl VerifyInputWithRecursion for MoveStructTag {
+    fn verify(&self, recursion_count: u8) -> anyhow::Result<()> {
+        verify_module_identifier(self.module.as_str())
+            .map_err(|_| anyhow::anyhow!("invalid struct tag: {}", self))?;
+        verify_identifier(self.name.as_str())
+            .map_err(|_| anyhow::anyhow!("invalid struct tag: {}", self))?;
```

Comment:
Could you make the error more specific, like `Invalid struct tag for module name`, `Invalid struct tag for function name`, etc.